lib/api/controllers/debugController.ts

```diff
@@ -0,0 +1,318 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2022 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { KuzzleRequest } from '../request';
+import { NativeController } from './baseController';
+import Inspector from 'inspector';
+import * as kerror from '../../kerror';
+import { DebugModule } from '../../types/DebugModule';
+import { JSONObject } from 'kuzzle-sdk';
+import get from 'lodash/get';
+
+const DEBUGGER_EVENT = 'kuzzle-debugger-event';
+
+type DebugModuleMethod = (params: JSONObject) => any;
+
+/**
+ * @class DebugController
+ */
+export class DebugController extends NativeController {
+  private inspector: Inspector.Session;
+
+  private debuggerStatus = false;
+
+  /**
+   * Map<eventName, Set<connectionId>>
+   */
+  private events = new Map<string, Set<string>>();
+
+  /**
+   * Map of functions from the DebugModules
+   */
+  private kuzzlePostMethods = new Map<string, DebugModuleMethod>();
+
+  /**
+   * List of DebugModule for DebugController
+   * Used to add new methods and events to the protocol
+   */
+  private modules: DebugModule[] = [];
```

Comment:
how this array is initialized ?

Reply:
This is a constant array, modules will be added in other PR like so:
```ts
private modules: DebugModule[] = [
   new DebugModule1(),
   new DebugModule2(),
   new EventMonitor(),
   ...
];
```